Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/JsonExportProvider.cs

```diff
@@ -12,20 +18,83 @@ public class JsonExportProvider : IExportProvider
         public IExportProviderConfiguration Configuration { get; }
         public ExportedTypeMetadata Metadata { get; set; }
 
+        private readonly JsonSerializer _serializer;
+        private StreamWriter _writer;
+
+
         public JsonExportProvider(Stream stream, IExportProviderConfiguration exportProviderConfiguration)
         {
+            if (exportProviderConfiguration is JsonProviderConfiguration jsonProviderConfiguration)
+            {
+                _serializer = JsonSerializer.Create(jsonProviderConfiguration.Settings);
+            }
+            else
+            {
+                _serializer = JsonSerializer.CreateDefault();
+            }
+
             _stream = stream;
             Configuration = exportProviderConfiguration;
         }
 
         public void WriteMetadata(ExportedTypeMetadata metadata)
         {
-            throw new System.NotImplementedException();
         }
 
         public void WriteRecord(object objectToRecord)
         {
-            throw new System.NotImplementedException();
+            EnsureWriterCreated();
+            FilterProperties(objectToRecord);
+            _serializer.Serialize(_writer, objectToRecord);
+            _writer.Flush();
+        }
+
+
+        private void EnsureWriterCreated()
+        {
+            if (_writer == null)
+            {
+                _writer = new StreamWriter(_stream);
+            }
+        }
+
+        private void FilterProperties(object obj, string baseMemberName = null)
+        {
+            var type = obj.GetType();
+
+            foreach (var property in type.GetProperties().Where(x => x.CanRead && x.CanWrite))
```

Comment:
Could we go not through all object properties (and links - there could be a lot of them), but to read only properties we get in Metadata?